### PR TITLE
QrData constructor for init from string representation

### PR DESCRIPTION
**Pull request by marektresnak**

### CODE REVIEW

File: src/Mews.FiscalChaining/Dto/Receipt.cs [hunk 1]
```diff
@@ -91,6 +125,12 @@ private byte[] GetValueBytes(long value)
             return orderedBytes.ToArray();
         }
 
+        private long GetValuesFromBytes(byte[] bytes)
+        {
+            var orderedBytes = BitConverter.IsLittleEndian ? bytes.Reverse() : bytes;
```
Comment (KaliCZ) [inline src/Mews.FiscalChaining/Dto/Receipt.cs]: Match

Comment (marektresnak) [reply src/Mews.FiscalChaining/Dto/Receipt.cs]: I tried to be consistent with the rest of the codebase. If we want stop using ternary operators (and possibly enforce some other code-style rules), I think we should do it in a separate pull request.

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 2]
```diff
@@ -51,5 +62,49 @@ private string ComputeValue()
...
+                return new CurrencyValue(val);
+            }
+            throw new ArgumentException("Currency value is not valid.");
+        }
+
+        private LocalDateTime ParseLocalDateTime(string value)
+        {
+            if (DateTime.TryParse(value, out DateTime val))
+            {
+                return new LocalDateTime(val, LocalDateTime.AustrianTimezone);
```
Comment (vitasamek) [inline src/Mews.FiscalChaining/Dto/QrData.cs]: is it only for Austria? Or why is this hardcoded?

File: src/Mews.FiscalChaining/Dto/Receipt.cs [hunk 3]
```diff
@@ -59,22 +85,30 @@ private EncryptedTurnover EncryptTurnover()
         {
             var sum = TaxData.Sum();
             var counter = (Turnover.Value + sum) * 100;
-            var registryReceiptId = RegisterIdentifier.Value + Number.Value;
-            var initializationVectorBytes = Encoding.UTF8.GetBytes(registryReceiptId);
-            var registryReceiptIdHash = Sha256(initializationVectorBytes);
-            var encryptedValue = AesCtr(registryReceiptIdHash, (long) counter, Key);
+            var valueBytes = GetValueBytes((long) counter);
```
Comment (vitasamek) [inline src/Mews.FiscalChaining/Dto/Receipt.cs]: no space (long)counter

File: src/Mews.FiscalChaining/Dto/Receipt.cs (hunk 1 above)
Comment (vitasamek) [inline src/Mews.FiscalChaining/Dto/Receipt.cs]: match

Comment (marektresnak) [reply src/Mews.FiscalChaining/Dto/Receipt.cs]: We don't have FuncSharp in this library.

File: src/Mews.FiscalChaining/Dto/QrData.cs (hunk 2 above)
Comment (marektresnak) [reply src/Mews.FiscalChaining/Dto/QrData.cs]: Not just in Austria. But in other set-ups, it doesn't really matter what timezone we use. We just need to be consistent.

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 4]
```diff
@@ -51,5 +62,49 @@ private string ComputeValue()
 
             return $"_{String.Join("_", fieldValues)}";
         }
+
+        private Receipt ParseReceiptValue(string value, byte[] key)
```
[the remark this answers is not in the document]
Comment (marektresnak) [reply src/Mews.FiscalChaining/Dto/QrData.cs]: I like ParseReceipt a bit better, but I can live with both.

[the remark this answers is not in the document]
Comment (ondrahermanek) [reply src/Mews.FiscalChaining/Dto/QrData.cs]: I am fine with "Parse" I am not fine with "Value". So "ParseValue" is OK for me.

[the remark this answers is not in the document]
Comment (ondrahermanek) [reply src/Mews.FiscalChaining/Dto/QrData.cs]: yes

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 5]
```diff
@@ -1,14 +1,25 @@
-﻿using System;
+﻿using Mews.FiscalChaining.Dto.Identifiers;
+using System;
+using System.Globalization;
+
 namespace Mews.FiscalChaining.Dto
 {
     public sealed class QrData
     {
+        public static readonly CultureInfo AustrianCultureInfo = CultureInfo.GetCultureInfo("de-AT");
```
Comment (siroky) [inline src/Mews.FiscalChaining/Dto/QrData.cs]: Make this a parameter.

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 6]
```diff
@@ -1,14 +1,25 @@
...
+
         public QrData(Receipt receipt)
         {
             Receipt = receipt;
             Value = ComputeValue();
         }
 
+        public QrData(string value, byte[] key)
```
Comment (siroky) [inline src/Mews.FiscalChaining/Dto/QrData.cs]: Create a third constructor with receipt and value and make these two constructors call it.

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 7]
```diff
@@ -51,5 +62,49 @@ private string ComputeValue()
 
             return $"_{String.Join("_", fieldValues)}";
         }
+
+        private Receipt ParseReceipt(string value, byte[] key)
```
Comment (siroky) [inline src/Mews.FiscalChaining/Dto/QrData.cs]: Move this to receipt class maybe?

File: src/Mews.FiscalChaining/Dto/QrData.cs [hunk 8]
```diff
@@ -51,5 +62,49 @@ private string ComputeValue()
...
+                    ParseCurrencyValue(values[8])
+                ),
+                encryptedTurnover: new EncryptedTurnover(values[9]),
+                certificateSerialNumber: new CertificateSerialNumber(values[10]),
+                chainValue: new ChainValue(values[11]),
+                key: key
+            );
+        }
+
+        private CurrencyValue ParseCurrencyValue(string value)
```
Comment (siroky) [inline src/Mews.FiscalChaining/Dto/QrData.cs]: Move these to corresponding classes as well and parametrize with culture

**Pull request merged**